js_modules/dagster-ui/packages/ui-core/src/assets/asset-checks/AssetChecksStatusSummary.tsx

```diff
@@ -51,26 +63,70 @@ const AssetCheckIconsOrdered: {
   },
 ];
 
+const titlePerCheckType = (type: AssetCheckIconType) => {
+  switch (type) {
+    case AssetCheckExecutionResolvedStatus.IN_PROGRESS:
+      return 'In progress';
+    case AssetCheckExecutionResolvedStatus.SKIPPED:
+      return 'Skipped';
+    case AssetCheckExecutionResolvedStatus.SUCCEEDED:
+      return 'Succeeded';
+    case 'NOT_EVALUATED':
+      return 'Not evaluated';
+    case 'WARN':
+      return 'Failed';
+    case 'ERROR':
+      return 'Failed';
+    default:
+      assertUnreachable(type);
+  }
+};
+
+const ChecksSummaryPopover = ({
+  type,
+  assetKey,
+  assetChecks,
+}: {
+  type: AssetCheckIconType;
+  assetKey: AssetKey;
+  assetChecks: AssetCheckLiveFragment[];
+}) => {
+  return (
+    <Box flex={{direction: 'column'}} style={{maxHeight: 300, overflowY: 'auto'}}>
+      <Box padding={{horizontal: 12, vertical: 8}} border="bottom">
+        <Subtitle2>{`${titlePerCheckType(type)} checks`}</Subtitle2>
+      </Box>
+      {assetChecks.map((check) => (
+        <CheckStatusRow key={check.name} assetCheck={check} assetKey={assetKey} />
+      ))}
+    </Box>
+  );
+};
+
+function statusFromCheck(check: AssetCheckLiveFragment): AssetCheckIconType {
+  const status = check.executionForLatestMaterialization?.status;
+  return status === undefined
+    ? 'NOT_EVALUATED'
+    : status === AssetCheckExecutionResolvedStatus.FAILED
+    ? check.executionForLatestMaterialization?.evaluation?.severity === AssetCheckSeverity.WARN
+      ? 'WARN'
+      : 'ERROR'
+    : status === AssetCheckExecutionResolvedStatus.EXECUTION_FAILED
+    ? 'ERROR'
+    : status;
+}
+
 export const AssetChecksStatusSummary = ({
   liveData,
   rendering,
+  assetKey,
 }: {
   liveData: LiveDataForNode;
   rendering: 'dag' | 'tags';
+  assetKey: AssetKey;
 }) => {
   const byIconType = countBy(liveData.assetChecks, (c) => {
-    const status = c.executionForLatestMaterialization?.status;
-    const value: AssetCheckIconType =
-      status === undefined
-        ? 'NOT_EVALUATED'
-        : status === AssetCheckExecutionResolvedStatus.FAILED
-        ? c.executionForLatestMaterialization?.evaluation?.severity === AssetCheckSeverity.WARN
-          ? 'WARN'
-          : 'ERROR'
-        : status === AssetCheckExecutionResolvedStatus.EXECUTION_FAILED
-        ? 'ERROR'
-        : status;
-    return value;
+    return statusFromCheck(c);
```

Comment:
Dumb javascript nit, but since `c` is passed straight through here and we don't need anything else from the surrounding scope, you could actually write `countBy(liveData.assetChecks, statusFromCheck)`

Also thanks for moving these out to helper methods, I think it's a nice improvement!

Reply:
ahh good call out, fixed